src/parser/syntax.rs

```diff
@@ -85,61 +89,125 @@ pub fn parse_symbol_limbs<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_
     }
 }
 
-pub fn parse_absolute_symbol<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_, F, Symbol> {
+fn intern_path<'a, F: LurkField>(
+    state: Rc<RefCell<State>>,
+    upto: LocatedSpan<&'a str>,
+    path: &[String],
+    keyword: Option<bool>,
+    create_unknown_packages: bool,
+) -> ParseResult<'a, F, SymbolRef> {
+    use nom::Err::Failure;
+    match keyword {
+        Some(keyword) => {
+            match state
+                .borrow_mut()
+                .intern_path(path, keyword, create_unknown_packages)
+            {
+                Ok(symbol) => Ok((upto, symbol)),
+                Err(e) => Err(Failure(ParseError::new(
+                    upto,
+                    ParseErrorKind::InterningError(format!("{e}")),
+                ))),
+            }
+        }
+        None => match state
+            .borrow_mut()
+            .intern_relative_path(path, create_unknown_packages)
+        {
+            Ok(symbol) => Ok((upto, symbol)),
+            Err(e) => Err(Failure(ParseError::new(
+                upto,
+                ParseErrorKind::InterningError(format!("{e}")),
+            ))),
+        },
+    }
```

Comment:
Isn't this just the following?
```rust
    match keyword {
        Some(keyword) => state
            .borrow_mut()
            .intern_path(path, keyword, create_unknown_packages),
        None => state
            .borrow_mut()
            .intern_relative_path(path, create_unknown_packages),
    }
    .map(|symbol| (upto, symbol))
    .map_err(|e| {
        Failure(ParseError::new(
            upto,
            ParseErrorKind::InterningError(format!("{e}")),
        ))
    })
```

Also, nit, the linear amount of recursive checks on keyword is unfortunate. You might consider `intern_path<'a, F: LurkField, const CREATE_UNKOWN: bool>(...`